application-gateway-logviewer-goaccess/azuredeploy.json

```diff
@@ -0,0 +1,183 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {    
+    "adminUsername": {
+      "type": "string",
+      "defaultValue": "ubuntu",
+      "metadata": {
+        "description": "Username you want to use for the VM the template creates"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password you want to use to log in to the VM"
+      }
+    },
+    "dnsNameForPublicIP": {
+      "type": "string",
```

Comment:
add a defaultValue using uniqueString()

Reply:
We are using this parameter to derive names of other resources in the template. Changing it to something which needs to be computed does not work when I try to use it's value in the variables. Can you give me some examples when we use any operation in parameter value and same parameter is used for deriving any variable names?

Reply:
depends on the error you're getting... common practice is to use something like:

[uniqueString(resourceGroup().id)]
or
[concat('dns', uniqueString(resourceGroup().id))]

Reply:
Using [concat('appgw', uniqueString(resourceGroup().id))] is failing for me as per the errors below:
---------------------------------------------------------------------------------------------------------------------------------
{"telemetryId":"152aba78-be18-4cfa-ba4a-5dacacf0382a","bladeInstanceId":"Blade_ce79d4ad4f1541a9a468ee8607824f54_0_0","galleryItemId":"Microsoft.Template","createBlade":"DeployToAzure","code":"InvalidTemplateDeployment","message":"The template deployment 'Microsoft.Template' is not valid according to the validation procedure. The tracking id is 'f1413cf9-5bbc-4297-a6a4-6d03e2f615ae'. See inner errors for details. Please see https://aka.ms/arm-deploy for usage details.","details":[{"code":"TemplateFailedPrevalidation","message":"The template request has multiple validation failures involving one or more Microsoft.Network resources. Please check details for information about each resource.","details":[{"code":"InvalidResourceName","message":"Resource name [concat('appgw', uniqueString(resourceGroup().id))]_nsg is invalid. The name can be up to 80 characters long. It must begin with a word character, and it must end with a word character or with '_'. The name may contain word characters or '.', '-', '_'."},{"code":"InvalidResourceName","message":"Resource name [concat('appgw', uniqueString(resourceGroup().id))]_vnet is invalid. The name can be up to 80 characters long. It must begin with a word character, and it must end with a word character or with '_'. The name may contain word characters or '.', '-', '_'."},{"code":"InvalidResourceName","message":"Resource name [concat('appgw', uniqueString(resourceGroup().id))]_vmnic is invalid. The name can be up to 80 characters long. It must begin with a word character, and it must end with a word character or with '_'. The name may contain word characters or '.', '-', '_'."}]}]}
---------------------------------------------------------------------------------------------------------------------------------